### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by ahadb802**
## Bookstore: add reducers and actions

- I have configured my Redux Store and write actions and reducer for adding and removing books.
- Added Redux Toolkit`(npm install react-redux @reduxjs/toolkit)`.
- Defined action types for adding and removing a book.
- Set the initial state to be an empty array of books.
- Export Action Creators for your actions.
- Write your reducer and export it as default.
- Define state changes for the actions that you created.
- In case of unknown action - return the current state.
- Define an action type for checking the status.

- Set the initial state to be an empty array of categories.
- Export Action Creators for your actions.
- Write your reducer and export it as default.
- For the check status action return a string "Under construction".
- In case of unknown action - return the current state.
- importing the necessary methods from Redux Toolkit.
- Combine both reducers into a root reducer by using configureStore function.

### CODE REVIEW

File: src/redux/books/book.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const ADD_A_BOOK = 'ADD_A_BOOK';
+const REMOVE_A_BOOK = 'REMOVE_A_BOOK';
+const LIST_BOOKS = 'LIST_BOOKS';
```
Comment (codepantha) [inline src/redux/books/book.js]: - [ ] Following the ducks pattern, your action types should be in the form `app/reducer/ACTION_TYPE` e.g `bookstore/books/ADD_A_BOOK`. Please make modification 👍

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const CHECK_STATUS = 'CHECK_STATUS';
+
```
Comment (codepantha) [inline src/redux/categories/categories.js]: - [ ] Following the ducks pattern, your action types should be in the form app/reducer/ACTION_TYPE e.g bookstore/categories/CHECK_STATUS. Please make modification 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @ahadb802 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added reducers and actions 📌 
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- NA
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**